### PR TITLE
fix: amplify_outputs for apple & amazon login method

### PR DESCRIPTION
**Pull request by bombguy**
## Problem

**Issue number, if available:**
fixes https://github.com/aws-amplify/amplify-backend/issues/1506
## Changes

Updating amplify_output.json to emit correct values for identity_providers.
```
AMAZON -> LOGIN_WITH_AMAZON
APPLE -> SIGN_IN_WITH_APPLE
```
**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: packages/auth-construct/src/construct.ts [hunk 1]
```diff
@@ -700,7 +700,7 @@ export class AmplifyAuth
       );
       result.oAuthMappings[authProvidersList.amazon] =
         external.loginWithAmazon.clientId;
-      result.providersList.push('AMAZON');
+      result.providersList.push('LOGIN_WITH_AMAZON');
```
Comment (edwardfoyle) [inline packages/auth-construct/src/construct.ts:703]: is there some way we can get the type system to fail incorrect values at build time here?

**Review by edwardfoyle — APPROVED** [review]

File: packages/auth-construct/src/construct.ts (hunk 1 above)
Comment (bombguy) [reply packages/auth-construct/src/construct.ts:703]: The best way to catch this in the future will be in some form of integration test, by reading amplify_config.json and comparing it to the config schema.

https://github.com/aws-amplify/amplify-backend/blob/main/packages/client-config/src/client-config-schema/schema_v1.json

Comment (edwardfoyle) [reply packages/auth-construct/src/construct.ts:703]: Perhaps we should go ahead and add that test? We could use [ajv](https://www.npmjs.com/package/ajv) to validate that the client config matches the schema

Comment (bombguy) [reply packages/auth-construct/src/construct.ts:703]: I can follow up and try adding a test for the JSON validation.

**Review by 0618 — APPROVED** [review]

**Pull request merged**